src/main/java/org/apache/maven/plugin/compiler/AbstractCompilerMojo.java

```diff
@@ -1172,6 +1172,8 @@ public void execute() throws MojoExecutionException, CompilationFailureException
                 try (Stream<Path> walk =
                         Files.walk(getGeneratedSourcesDirectory().toPath())) {
                     walk.sorted(Comparator.reverseOrder()).map(Path::toFile).forEach(File::delete);
+                    // MCOMPILER-567: recreate directory
+                    Files.createDirectories(getGeneratedSourcesDirectory().toPath());
```

Comment:
Instead of creating directory we can detect not existing in plexus-compiler - what do you think

Reply:
Sure, that should work too.

Reply:
In any case, it does not harm to create the directory too.